src/transforms/operationTransforms.ts

```diff
@@ -242,36 +255,64 @@ export function transformOperation(
       ? operation.apiVersions.map(v => v.version)
       : [],
     description: metadata.description,
-    request: transformOperationRequest(operation.request),
-    responses: responses.map(response =>
-      transformOperationResponse(response as SchemaResponse)
-    )
+    request,
+    responses,
+    mediaTypes
   };
 }
 
-export async function transformOperationGroups(codeModel: CodeModel) {
-  return codeModel.operationGroups.map(transformOperationGroup);
+export function transformOperationGroups(
+  codeModel: CodeModel
+): Promise<OperationGroupDetails[]> {
+  return Promise.all(codeModel.operationGroups.map(transformOperationGroup));
 }
 
-export function transformOperationGroup(
+export async function transformOperationGroup(
   operationGroup: OperationGroup
-): OperationGroupDetails {
+): Promise<OperationGroupDetails> {
   const metadata = getLanguageMetadata(operationGroup.language);
   const isTopLevel = !metadata.name;
   const name = normalizeName(
     metadata.name || TOPLEVEL_OPERATIONGROUP,
     NameType.Property
   );
+
+  const operations = await Promise.all(
+    operationGroup.operations.map(
+      async operation => await transformOperation(operation, name)
```

Comment:
Minor but I don't think async/await is needed here since you're doing a Promise.all. Similar general feedback that the returned promise from tranformOperation would be unwrapped by await then rewrapped in a new promise.

This change should work since transformOperation returns a promise
```suggestion
      operation => transformOperation(operation, name)
```